### DISCUSSION TITLE
Not a registered object

### DISCUSSION THREAD

**Discussion by Emmanuel-324** [idaholab/moose#27623] [Q&A Modules: General]
I get the error below when I run my input script. This was not the case before I installed my MOOSE again on my HPC cluster.
```
A 'RankTwoScalarAux' is not a registered object.check_inp(moose_srv)
If you are trying to find this object in a dynamically loaded library, make sure thatcheck_inp(moose_srv)
the library can be found either in your "Problem/library_path" parameter or in thecheck_inp(moose_srv)
MOOSE_LIBRARY_PATH environment variable.
```

**Comment by Rahim-Habibi** [comment]
it means that 'RankTwoScalarAux' does not exist in your moose directory. 
probably you need to check the directory: moose/modules/tensor_mechanics/src/auxkernels
if it is not there, you can get it from moose page and put it in right directory, then compile it.

**Reply by GiudGiud** [reply]
This auxkernel has been around since 2017 so I dont think that's the problem.

I think the executable is not compiled with the solid mechanics module maybe? Can you check the Makefile for your app and make sure the SOLID_MECHANICS field is set to yes?
Once that's done, you ll need to clean and recompile

> if it is not there, you can get it from moose page and put it in right directory, then compile it.

we dont recommend doing that.
Instead update MOOSE.

**Reply by GiudGiud** [reply]
when you clean, use `git clean -xfd` and use it in the top folder of the app

**Reply by Emmanuel-324** [reply]
I set all my modules to yes when I compiled my application. Nevertheless, I am going to do it again.

**Reply by GiudGiud** [reply]
The cleaning before compiling part is important too unfortunately

**Reply by Emmanuel-324** [reply]
Yes, I cleaned my app folder before compiling it. For the record, I used this in compiling: 
cd ~/projects/YourAppName
make clobberall
make -j 6
./run_tests -j 6

**Reply by Emmanuel-324** [reply]
I run into this problem after doing that:
```
WASP does not seem to be available.
Make sure to either run scripts/update_and_rebuild_wasp.sh in your MOOSE directory,
or set WASP_DIR to a valid WASP install
make: *** [wasp_submodule_status] Error 1
make: *** Waiting for unfinished jobs....
Rebuilding symlinks in /home/emmanuel324/projects/moose/framework/build/header_symlinks
(python-stack) [emmanuel324@tinkercliffs1 belson]$ ./run_tests -j 6
```

**Reply by GiudGiud** [reply]
you forgot to activate the moose environment

**Reply by GiudGiud** [reply]
note that make clobberall is not cleaning enough. You need to bring out the big deep clean guns with `git clean -Xfd`

**Reply by Emmanuel-324** [reply]
I did git clean-xfd in my app folder

(python-stack) [emmanuel324@tinkercliffs1 belson]$ git clean -xfd
Removing .libs/
Removing .previous_test_results.json
Removing belson-opt
Removing belson.yaml
Removing build/
Removing key
Removing key.pub
Removing lib/
Removing restartkks/input_slip_sys.txt
Removing restartkks/input_slip_sys_bcc48.txt
Removing restartkks/kks_multiphase_largescale_large_out.e-s183
Removing restartkks/restartkks_soriginal.out
Removing src/.libs/
Removing src/main.x86_64-pc-linux-gnu.opt.lo
Removing src/main.x86_64-pc-linux-gnu.opt.lo.d
Removing test/lib/
Removing test/src/base/.libs/
Removing test/src/base/belsonTestApp.x86_64-pc-linux-gnu.opt.lo
Removing test/src/base/belsonTestApp.x86_64-pc-linux-gnu.opt.lo.d
Removing test/tests/kernels/simple_diffusion/simple_diffusion_out.e

**Reply by GiudGiud** [reply]
ok good. be careful with this btw, it removes absoltely everything that has not been committed

**Reply by Emmanuel-324** [reply]
With the commits, what is the best way to go about it, anytime I try to commit my changes I get the output as it is not properly formatted

**Reply by GiudGiud** [reply]
I do
```
git add <the files you care about>
git commit -m "a commit message"
```
but there's plenty of options

what error do you get?

**Reply by Emmanuel-324** [reply]
I usually get something about git clang not being git clang not being a command and my file not being properly formatted.

**Reply by GiudGiud** [reply]
need to install git-clang-format then. It can be done from conda

**Reply by Emmanuel-324** [reply]
What is the proper way to recompile after making some changes in the source file?

**Reply by GiudGiud** [reply]
Go in the folder with the Makefile and type 'make -j 6' there

**Reply by Emmanuel-324** [reply]
Yes, I thought so too. Thanks. I was adding clobberall and  that was deleting my wasp all the time.
